src/test/java/org/jenkinsci/plugins/mesos/integration/MesosSlaveLifecycleTest.java

```diff
@@ -53,4 +53,36 @@ public void testAgentLifecycle(TestUtils.JenkinsRule j) throws Exception {
 
     assertThat(Jenkins.getInstanceOrNull().getNodes(), hasSize(0));
   }
+
+  @Test
+  public void testComputerNodeTermination(TestUtils.JenkinsRule j) throws Exception {
+    MesosCloud cloud = new MesosCloud("mesos", mesosCluster.getMesosUrl(), j.getURL().toString());
+
+    MesosSlave agent = (MesosSlave) cloud.startAgent().get();
+    agent.waitUntilOnlineAsync().get();
+
+    assertThat(agent.isRunning(), is(true));
+    assertThat(agent.getComputer().isOnline(), is(true));
+
+    MesosSlave shouldBeParent = (MesosSlave) agent.getComputer().getNode();
+    shouldBeParent.terminate();
+    await().atMost(5, TimeUnit.MINUTES).until(agent::isKilled);
```

Comment:
This slows down tests. What do you think about reducing the idle time to a few seconds and verify that?

Reply:
sure it shouldnt have an effect in my opinion since if agent isKilled or isRunning does not return within a few seconds there is probably an issue, I will lower idle time to 30 sec